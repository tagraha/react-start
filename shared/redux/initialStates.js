export const postMock = {
  asyncPostExample: {
    body: '',
    id: 0,
    title: '',
    userId: 0,
  },
};

export const counterMock = {
  counterValue: 0,
};

export const errorMock = {};
