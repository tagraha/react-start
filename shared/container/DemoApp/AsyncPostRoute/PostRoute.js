import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withJob } from 'react-jobs';
import Helmet from 'react-helmet';
import { invokeIncrement } from './../../../redux/modules/counter';
import { loadPost } from './../../../redux/modules/post';

// internal Components
import Counter from './../../../components/Demo/Counter';

const PostRoute = (props) => {
  const { counter, incrementAction, post } = props;

  const increment = () => {
    incrementAction();
  };

  return (
    <div>
      <Helmet>
        <title>Async Post Example</title>
      </Helmet>

      <h3>Redux</h3>
      <h4>
        async example{' '}
        <small>
          (fetch from: https://jsonplaceholder.typicode.com/posts/1)
        </small>
      </h4>
      <p>{post.asyncPostExample.body}</p>

      <p>
        Redux counter value: <Counter count={counter.counterValue} />
      </p>
      <button type="button" onClick={increment}>
        Increment
      </button>
    </div>
  );
};

const mapStateToProps = (state) => ({
  counter: state.counter,
  post: state.post,
});

const mapActionsToProps = {
  incrementAction: invokeIncrement,
  asyncDemo: loadPost,
};

PostRoute.propTypes = {
  counter: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  post: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  incrementAction: PropTypes.func.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default compose(
  connect(mapStateToProps, mapActionsToProps),
  withJob({
    work: (
      { match, post, asyncDemo }, // eslint-disable-line
    ) => asyncDemo(), // Execute the redux-thunk powered action that returns a Promise and fetches the post.
    // Any time the post id changes we need to trigger the work.
    shouldWorkAgain: (prevProps, nextProps) =>
      // eslint-disable-line
      prevProps.post.asyncPostExample.id !== nextProps.post.asyncPostExample.id,
  }),
)(PostRoute);
