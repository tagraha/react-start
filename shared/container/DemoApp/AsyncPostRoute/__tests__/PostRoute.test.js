/* @flow */
/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { shallow } from 'enzyme';
import PostRoute from '../PostRoute';
import configureStore from 'redux-mock-store';
import { postMock } from './../../../../redux/initialStates';

describe('<PostRoute />', () => {
  test('renders', () => {
    const mockStore = configureStore();
    const store = mockStore(postMock);
    const wrapper = shallow(<PostRoute store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
