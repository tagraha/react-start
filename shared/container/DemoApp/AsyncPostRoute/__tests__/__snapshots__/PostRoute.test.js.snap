// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostRoute /> renders 1`] = `
<ContextProvider
  value={null}
>
  <WithJob(PostRoute)
    asyncDemo={[Function]}
    incrementAction={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
